spellcheck.pl

```diff
@@ -135,30 +140,39 @@ sub spellcheck_key_pressed
 
     return unless Irssi::settings_get_bool('spellcheck_enabled');
 
-    # I know no way to *mark* misspelled words in the input line,
-    # that's why there's no spellcheck_print_suggestions -
-    # because printing suggestions is our only choice.
-
     # hide correction window when message is sent
     if (chr($key) =~ /\A[\r\n]\z/ && $correction_window) {
         $correction_window->command("window hide $window_name");
+        if (Irssi->can('gui_input_clear_extents')) {
+            Irssi::gui_input_clear_extents(0, 9999);
+        }
+    }
+
+    # get current inputline
+    my $inputline = Irssi::parse_special('$L');
+    if (Irssi::settings_get_str('term_charset') =~ /utf-?8/i) {
+        Encode::_utf8_on($inputline);
```

Comment:
…but maybe it would be better to just decode the input line using `term_charset` encoding? Then you wouldn't need any ifs or regexps.

Reply:
I don't think that's economic